src/servers/main.cc

```diff
@@ -247,31 +282,105 @@ StartHttpService(nvidia::inferenceserver::InferenceServer* server)
   return std::move(service);
 }
 
+std::unique_ptr<nvidia::inferenceserver::GRPCServer>
+StartMultipleGrpcService(nvidia::inferenceserver::InferenceServer* server, std::string endpoint_name, int32_t endpoint_port)
+{
+  std::unique_ptr<nvidia::inferenceserver::GRPCServer> service;
+  nvidia::inferenceserver::Status status =
+      nvidia::inferenceserver::GRPCServer::CreateUniqueEndpointPorts(server, endpoint_name, endpoint_port, &service);
+  if (status.IsOk()) {
+    status = service->Start();
+  }
+
+  if (!status.IsOk()) {
+    service.reset();
+  }
+
+  return std::move(service);
+}
+
+std::unique_ptr<nvidia::inferenceserver::HTTPServer>
+StartMultipleHttpService(nvidia::inferenceserver::InferenceServer* server, std::string endpoint_name, int32_t endpoint_port)
+{
+  std::unique_ptr<nvidia::inferenceserver::HTTPServer> service;
+  nvidia::inferenceserver::Status status =
+      nvidia::inferenceserver::HTTPServer::CreateUniqueEndpointPorts(
+          server, endpoint_name, endpoint_port, http_thread_cnt_, &service);
+  if (status.IsOk()) {
+    status = service->Start();
+  }
+
+  if (!status.IsOk()) {
+    service.reset();
+  }
+
+  return std::move(service);
+}
+
 bool
 StartEndpoints(nvidia::inferenceserver::InferenceServer* server)
 {
   LOG_INFO << "Starting endpoints, '" << server->Id() << "' listening on";
 
+  // Start normally if common port else spawn different servers for different endpoints
```

Comment:
We're not sure if/what additional overhead is introduced by having multiple sevice instances (grpc or http) so I'd like to minimize as much as possible. So can you change the existing Create functions to take all 4 ports and be smart to look at those ports and start the minimum number of services needed. This will keep the same behavior as currently when all 4 ports are the same, but will (for example) only start one additional service is only one port is different.